CMakeLists.txt

```diff
@@ -65,6 +65,13 @@ if(ENABLE_LOG_DEBUG AND ENABLE_SYSTEM_STATS)
 	add_definitions(-DMBED_HEAP_STATS_ENABLED)
 endif(ENABLE_LOG_DEBUG AND ENABLE_SYSTEM_STATS)
 
+# Firmware version
+option(FIRMWARE_VERSION "Set firmware version" "0.0.0")
+
+if(FIRMWARE_VERSION)
```

Comment:
as you give the initial value `0.0.0` to `FIRMWARE_VERSION`, the condition is always true.
the `if` can be removed

Reply:
ok